lib/codeowners/cli/main.rb

```diff
@@ -61,25 +60,64 @@ def write_codeowners
       end
 
       def suggest_add_to_codeowners(file)
-        return unless yes?("File added: #{file.inspect}. Add owner to the CODEOWNERS file?")
+        return if @quit
 
-        owner = ask('File owner(s): ')
-        new_line = create_new_pattern(file, owner)
-
-        unless new_line.pattern?
-          puts "#{owner.inspect} is not a valid owner name."
-          return
+        case add_to_codeowners_dialog(file)
+        when 'y' then add_to_codeowners(file)
+        when 'i' then return
+        when 'q' then @quit = true
         end
+      end
+
+      def add_to_codeowners_dialog(file)
+        ask(<<~QUESTION, limited_to: %w[y i q])
+          File added: #{file.inspect}. Add owner to the CODEOWNERS file?
+          (y) yes
+          (i) ignore
+          (q) quit
+        QUESTION
+      end
+
+      def add_to_codeowners(file)
+        new_line = create_new_pattern(file)
 
         subgroups = @checker.main_group.subgroups_owned_by(new_line.owner)
         add_pattern(new_line, subgroups)
 
         @codeowners_changed = true
       end
 
-      def create_new_pattern(file, owner)
-        line = "#{file} #{owner}"
-        Codeowners::Checker::Group::Line.build(line)
+      def create_new_pattern(file)
+        list_owners
+        loop do
+          owner = new_owner
+          line = "#{file} #{owner}"
+          new_line = Codeowners::Checker::Group::Line.build(line)
+
+          return new_line if new_line.pattern?
+
+          puts "#{owner.inspect} is not a valid owner name. Try again." unless new_line.pattern?
+        end
+      end
+
+      def list_owners
+        puts 'Owners:'
+        sorted_owners = @checker.main_group.owners.sort
+        sorted_owners.each_with_index { |owner, i| puts "#{i + 1} - #{owner}" }
+        puts "Choose owner, add new one or leave empty to use #{@config.default_owner.inspect}."
+      end
+
+      def new_owner
+        owner = ask('New owner: ')
+        sorted_owners = @checker.main_group.owners.sort
+
+        if owner.to_i.between?(1, sorted_owners.length)
+          sorted_owners[owner.to_i - 1]
+        elsif owner.empty?
+          @config.default_owner
+        else
+          owner
```

Comment:
we should validate at least it contains `@` as it's always required for either handle or email

also in light of #39 maybe we should extract this validation method to a common place as it is being validated few times? (https://github.com/toptal/codeowners-checker/blob/master/lib/codeowners/checker/group/pattern.rb#L15)

Reply:
We are validating the owner using the method you are referring to.  When the owner is not valid the line is not recognized as a pattern and we are asking for the name again.
https://github.com/toptal/codeowners-checker/blob/5e2389013cc360d4279d48519c05e20910faa5a6/lib/codeowners/cli/main.rb#L97
Where do you think we should move the validation?